src/wallet/wallet.cpp

```diff
@@ -3027,9 +3027,27 @@ int64_t CWallet::GetOldestKeyPoolTime()
     if (setKeyPool.empty())
         return GetTime();
 
-    // load oldest key from keypool, get time and return
     CKeyPool keypool;
     CWalletDB walletdb(strWalletFile);
+
+    if (IsHDEnabled() && CanSupportFeature(FEATURE_HD_SPLIT))
+    {
+        // if HD & HD Chain Split is enabled, response max(oldest-internal-key, oldest-external-key)
+        int64_t now = GetTime();
+        int64_t oldest_external = now, oldest_internal = now;
+
+        for(const int64_t& id : setKeyPool)
+        {
+            if (!walletdb.ReadPool(id, keypool))
+                throw runtime_error(std::string(__func__) + ": read failed");
+            if (keypool.fInternal && keypool.nTime < oldest_internal)
+                oldest_internal = keypool.nTime;
+            else if (!keypool.fInternal && keypool.nTime < oldest_external)
+                oldest_external = keypool.nTime;
+        }
+        return std::max(oldest_internal, oldest_external);
+    }
+    // load oldest key from keypool, get time and return
```

Comment:
We still need the `FEATURE_HD_SPLIT` special case here. If don't want to return just the oldest key in that case. 
Reason:
* Assume people use GetOldestKeyPoolTime() to determine if it's time for a new backup (difficult question when it's HD anyways, but that actually what @TheBlueMatt came up in https://github.com/bitcoin/bitcoin/pull/9294/files#r96107555)
* Assume your key in the keypool at index 0 (=oldest key) is an external key.
* Now assume you call `getrawchangeaddress` multiple times and you only and completely refill your internal keypool.
* In that case, you want to return the oldest internal key because its very likely newer then your oldest external key.

But maybe I'm wrong.